tests/requestparsing/basicauth_test.go

```diff
@@ -0,0 +1,117 @@
+package requestparsing
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/google/go-safeweb/testing/requesttesting"
+)
+
+type basicAuthWant struct {
+	headers  map[string][]string
+	ok       bool
+	username string
+	password string
+}
+
+var basicAuthTests = []struct {
```

Comment:
Define these inside the test function.

Reply:
Both `basicAuthWant` and `basicAuthTests`? 
Similar for all table tests?

Reply:
Yup.